source/common/quic/envoy_quic_session_cache.cc

```diff
@@ -0,0 +1,168 @@
+#include "common/quic/envoy_quic_session_cache.h"
+
+namespace Envoy {
+namespace Quic {
+namespace {
+
+constexpr size_t MaxSessionCacheEntries = 1024;
+
+// Returns false if the SSL |session| doesn't exist or it is expired at |system_time|.
+bool isSessionValid(SSL_SESSION* session, SystemTime system_time) {
+  if (session == nullptr) {
+    return false;
+  }
+  const time_t now = std::chrono::system_clock::to_time_t(system_time);
+  if (now < 0) {
+    return false;
+  }
+  const uint64_t now_u64 = static_cast<uint64_t>(now);
+  const uint64_t session_time = SSL_SESSION_get_time(session);
+  const uint64_t session_expiration = session_time + SSL_SESSION_get_timeout(session);
```

Comment:
How costly are there two functions? Worth latching these two in Entry?

Reply:
They're very cheap, just dereferencing the session pointer.